### DISCUSSION TITLE
Calculate number of personagents in RegionAgents

### DISCUSSION THREAD

**Discussion by techno001** [mesa/mesa#2482] [Q&A]
I have mesa-geo project with RegionAgents as city boundaries and person agents, residing in the city-boundary. Person agent move to more suitable locations as in sugar-spice landscape example. After every step, I need to calculate the number of agents in each RegionAgent. Is there a way to do it?

**Comment by wang-boyu** [comment]
Yes there should be more than one way to do it.

1. If the logic to add and remove agents from regions are similar to our geo_schelling_points example: https://github.com/projectmesa/mesa-examples/blob/59106a81535548de67c2667a9990734778e46e69/gis/geo_schelling_points/geo_schelling_points/agents.py#L64-L74, then you could simply have a `num_agents` attribute in the RegionAgent, which gets updated when a person agent is added/removed.
2. Otherwise there's a `get_agents_as_GeoDataFrame` method in GeoSpace: https://github.com/projectmesa/mesa-geo/blob/071724056c7670a80b0454f9ec44e4a5cffa380b/mesa_geo/geospace.py#L227. Using this method you could have a `person_gdf` by calling `model.space.get_agents_as_GeoDataFrame(agent_cls=PersonAgent)` and similarly a `region_gdf`. Then the number of agents in each region can be computed by performing a spatial join with GeoPandas. However this might be much slower than method 1) because you will need to extract agents as geodataframe and do spatial join at each model step.
